modules/storage/storage.go

```diff
@@ -82,3 +141,22 @@ func (v *VolMounts) Propagate(svc []PropagationType) []VolMounts {
 
 	return vl
 }
+
+// ConvertVolumeSource function to convert from a VolumeSource to a corev1.VolumeSource
+func ConvertVolumeSource(v *VolumeSource) corev1.VolumeSource {
```

Comment:
an alternative would be to do it like I did it for convert corev1.ServiceSpec to OverrideServiceSpec here https://github.com/openstack-k8s-operators/lib-common/blob/main/modules/common/service/service.go#L215 . the only benefit would be that you'd not have to update this func when there is a new VolumeSource

Reply:
I agree, this would be nice and a potential (future) update would be reduced to an additional field in the `VolumeSource` struct.

Reply:
My latest commit to the PR adds the new function 061cc8a

Reply:
awesome! thank you @dprince!